pkg/api/controller.go

```diff
@@ -1626,23 +1626,36 @@ func (c *Controller) validateStorageNamespace(storageNamespace string) error {
 }
 
 func (c *Controller) ensureStorageNamespace(ctx context.Context, storageNamespace string) error {
-	const (
-		dummyKey  = "dummy"
-		dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
-	)
+	dummyKey := fmt.Sprintf("%s/dummy", c.Config.Committed.BlockStoragePrefix)
+	const dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
 
-	obj := block.ObjectPointer{
+	// check if dummy file exists in previous lakeFS versions at the root of the storage namespace
+	objLegacyLocation := block.ObjectPointer{
 		StorageNamespace: storageNamespace,
 		IdentifierType:   block.IdentifierTypeRelative,
-		Identifier:       dummyKey,
+		Identifier:       "dummy",
 	}
 	objLen := int64(len(dummyData))
-	if _, err := c.BlockAdapter.Get(ctx, obj, objLen); err == nil {
-		return fmt.Errorf("found lakeFS objects in the storage namespace(%s): %w",
+
+	if _, err := c.BlockAdapter.Get(ctx, objLegacyLocation, objLen); err == nil {
```

Comment:
YIKES!  As Mitch Hedberg would say,
> _I used to leak a file descriptor.  I still do.  But I used to, too.

Consider using `.Exists()` (which doesn't test readability), or just closing (which probably also tests readability):
```suggestion
	if exists, err := c.BlockAdapter.Exists(ctx, objLegacyLocation, objLen); err == nil {
```
or
```suggestion
	if s, err := c.BlockAdapter.Get(ctx, objLegacyLocation, objLen); err == nil {
		s.Close()
```

Reply:
Good 👁️ you have!